fix: automatic answer if TTS = "don't speak"/field is empty

david-allison opened the pull request:
The supplied listener wasn't called if the TTS wasn't invoked

We reorder saving to `metaDB` in case the automatic display answer were blocking

Fixes #9170
Fixes #9630
Related #4609

## How Has This Been Tested?
On my Android 11

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

david-allison committed:
refactor: extract mSpeakText to mTTs.isEnabled

david-allison committed:
refactor: rm ReadText reference from CardViewer

david-allison committed:
nf: typo

david-allison committed:
fix(auto-display-answer): continue if TTS empty

If the back side of a note is empty, the completion
listener should be called immediately

Fixes 9630

mikehardy reviewed (commented):
I know it's draft, but seems good, just paranoia questions as noted

david-allison committed:
fix: automatic answer if TTS = "don't speak"

The supplied listener wasn't called if the TTS wasn't invoked

We reorder saving to metaDB in case the automatic display answer were
blocking

We return from `textToSpeech`: we couldn't call the listener from there
as it's called in a loop and would call the listener multiple times

Fixes 9170

mikehardy reviewed (approved):
Nice!
